src/m365/tenant/commands/auditlog/auditlog-report.ts

```diff
@@ -0,0 +1,261 @@
+      url: auditURL,
+      headers: {
+        accept: 'application/json;'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<AuditlogReport[]>(requestOptions);
+  }
+  
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-c, --contentType <contentType>',
+        description: 'Audit content type of logs to be retrieved, should be one of the following: AzureActiveDirectory, Exchange, SharePoint, General, DLP'
+      },
+      {
+        option: '-s, --startTime [startTime]',
+        description: 'Start time of logs to be retrieved. Start time and end time must both be specified (or both omitted) and must be less than or equal to 24 hours apart.'
+      },
+      {
+        option: '-e, --endTime [endTime]',
+        description: 'End time of logs to be retrieved. Start time and end time must both be specified (or both omitted) and must be less than or equal to 24 hours apart.'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if ((<any>AuditContentTypes)[args.options.contentType] === undefined) {
+      return `${args.options.contentType} is not a valid value for the contentType option. Allowed values are ${Object.keys(AuditContentTypes).join(' | ')}`;
+    }
+
+    if (args.options.startTime || args.options.endTime) {
+      if ((args.options.startTime && !(args.options.endTime)) || (!(args.options.startTime) && args.options.endTime)) {
+        return `startTime and endTime must both be specified (or both omitted)`;
+      }
+
+      const parsedStartTime = Date.parse(args.options.startTime as string)
+      if (args.options.startTime && !(!parsedStartTime) !== true) {
+        return `${args.options.startTime} is not a valid startTime. Provide the date in one of the following formats:
+      'YYYY-MM-DD'
+      'YYYY-MM-DDThh:mm'
+      'YYYY-MM-DDThh:mmZ'
+      'YYYY-MM-DDThh:mm±hh:mm'`;
+      }
+
+      const parsedEndTime = Date.parse(args.options.endTime as string)
+      if (args.options.endTime && !(!parsedEndTime) !== true) {
+        return `${args.options.endTime} is not a valid endTime. Provide the date in one of the following formats:
+      'YYYY-MM-DD'
+      'YYYY-MM-DDThh:mm'
+      'YYYY-MM-DDThh:mmZ'
+      'YYYY-MM-DDThh:mm±hh:mm'`;
+      }
+
+      if (parsedStartTime && parsedEndTime) {
+        const StartEndDateDifference: number = (parsedEndTime - parsedStartTime) / (1000 * 60 * 60 * 24);
+        const StartdateTodayDifference: number = (new Date().getTime() - parsedStartTime) / (1000 * 60 * 60 * 24);
+        if (StartEndDateDifference != 1 || StartdateTodayDifference > 7) {
+          return `startTime and endTime must be less than or equal to 24 hours apart, with the start time prior to end time and start time no more than 7 days in the past`;
+        }
```

Comment:
If I understand the API correctly we want to allow persons to search within a day. With this limitation we can't do it.

I cannot perform for example such a search: 2020-12-29T15:00:00 and 2020-12-29T16:00:00.

I think this part should be changed to ((StartEndDateDifference > 0 && StartEndDateDifference <= 1) || StartdateTodayDifference > 7)

Reply:
That is totally correct @plamber . So silly mistake from my end.. My Bad... 🤦. Thanks for the find.

I had not even thought about the time details, since I was focusing only on dates and their difference. I will modify the code and will push it again..